### DISCUSSION TITLE
Returning List[response_model] doesn't respect `response_model_exclude_unset

### DISCUSSION THREAD

**Discussion by rubyfin** [fastapi/fastapi#9002] [Questions] [answered]
The 'response_model_exclude_unset' flag does not exclude fields that are set to Optional for a @app.get() endpoint.

It looks like this bug reappeared: https://github.com/tiangolo/fastapi/issues/1063.
Steps to reproduce are exactly the same. 

**Environment**
OS: Linux
FastAPI Version: 0.63
Python version: 3.7.3

**Comment by falkben** [comment]
I realize you are following up on a previous issue but can you follow the issue template so we don't have to searching back through other issues to find how to reproduce?

For instance, I cannot reproduce from the previous issue. See below. Test passes for me.

```python
from typing import List

import pydantic
from fastapi import FastAPI
from fastapi.testclient import TestClient

app = FastAPI()


class Item(pydantic.BaseModel):
    foobar: str = "foobar"
    foo: str
    bar: str = None


ALL_ITEMS = [Item(foo="foo0"), Item(foo="foo1"), Item(foo="foo2")]


@app.get("/items", response_model=List[Item], response_model_exclude_unset=True)
def get_all_items():
    return ALL_ITEMS


@app.get("/items/{item_index}", response_model=Item, response_model_exclude_unset=True)
def get_item(item_index: int):
    return ALL_ITEMS[item_index]


client = TestClient(app)


def test_items():
    resp = client.get("/items")
    assert resp.json() == [{"foo": "foo0"}, {"foo": "foo1"}, {"foo": "foo2"}]
```

**Comment by rubyfin** [comment]
Will try to do from now on. 

Interesting, I tried the sample code used in the original bug reported, and it does work.

It does not seem to work when a response model is used to filter/validate the data fetched by pandas read_sql() method.
To reproduce it: 

Step1: create a simple db table that has the same fields as the following model, with all set to NULL except for foo1, foo2 and foo3 that should have some values:

```
class FooResponse(BaseModel):
    foo1: str
    foo2: str
    foo3: float
    foo4: Optional[str]
    foo5: Optional[str]
    foo6: Optional[float]
    foo7: Optional[float]
    foo8: Optional[float]
    foo9: Optional[float]
```

Step2: use the following code, just create a connection string to your sample database:

```
from typing import Optional

import pandas as pd
from fastapi import FastAPI
from fastapi.responses import ORJSONResponse
from pydantic import BaseModel
from starlette.responses import Response
from sqlalchemy import create_engine

db_conn = create_engine('mssql+pyodbc:///?odbc_connect={}'.format(conn_str), connect_args={'autocommit': True})

tags_metadata = [
    {
        "name": "test-bug",
        "description": "test a bug.",
    },
]

app = FastAPI(
    openapi_tags=tags_metadata,
    title="Buggy feature",
    description="Test a bug",
)

class FooResponse(BaseModel):
    foo1: str
    foo2: str
    foo3: float
    foo4: Optional[str]
    foo5: Optional[str]
    foo6: Optional[float]
    foo7: Optional[float]
    foo8: Optional[float]
    foo9: Optional[float]


@app.get("/test-bug",  response_class=ORJSONResponse, response_model=FooResponse, response_model_exclude_unset=True)
async def test_bug():
    _df = pd.read_sql(
        sql='TestABugTable',
        con=db_conn
    )
    return Response(_df.to_json(orient='records'))

```

I use pandas 0.25.3 and MS SQL database, but I suspect it won't be dependent on those.

**Comment by sonowz** [comment]
I'm not sure if this is related, yet I found out that if `resonse_model` field is not explicitly given, then `response_model_exclude_unset=True` doesn't work.
In my case, the `response_model` was `List[MyPydanticModel]`.

**Comment by hypogealgaol** [comment]
Can also confirm the same behavior as @sonowz

**Comment by YuriiMotov** [comment, marked as answer]
The problem is in the code of endpoint function.

```
@app.get("/test-bug",  response_class=ORJSONResponse, response_model=FooResponse, response_model_exclude_unset=True)
async def test_bug():
    _df = pd.read_sql(sql='TestABugTable', con=db_conn)

    return Response(_df.to_json(orient='records'))  # !!!!!
```

When you return an instance of `Response`, FastAPI doesn't perform response validation. So, any `response_model_***` flags don't take effect.

Read more: https://fastapi.tiangolo.com/advanced/response-directly/
